### ISSUE TITLE
MTConnect nodeset compiler

### ISSUE THREAD

**Issue by marioricardojr** [open62541/open62541#1752]
Hi, i having difficult compiling mtconnect node set
i follow plcopen tutorial(https://open62541.org/doc/current/nodeset_compiler.html), but i could not compile.
have some tutorial? thanks

**Comment by Stasik0** [comment]
error log would be helpful

**Comment by Pro** [comment]
@marioricardojr without additional info we can not help you. Please fill out the template for the issue, which we provide when you click on new issue. Also read through this: https://stackoverflow.com/help/mcve

**Comment by Pro** [comment]
The issue here is that you are including the DI nodeset, while it is not needed by the MTConnect nodeset. Probably the nodeset compiler is then using NS IDX 1 for the DI nodeset, also within MTConnect, and thus `ns=1;i=9998` does not exist. 

So it should be enough if you execute the following (which works on my machine):
```
python tools/nodeset_compiler/nodeset_compiler.py
--internal-headers
--types-array=UA_TYPES
--types-array=UA_TYPES
--existing /open62541/deps/ua-nodeset/Schema/Opc.Ua.NodeSet2.xml
--xml /open62541/deps/ua-nodeset/MTConnect/Opc.Ua.MTConnect.NodeSet2.xml
/open62541/src_mtconnect/ua_namespace_mtconnect
```

**Comment by Pro** [comment]
Unfortunately we are not wizards... (Even though it would sometimes be really helpful)
How should we be able to help you, If you do not post the error you are getting? Also please use the Github Markdown formatting, or at least the corresponding buttons in the issue editor to post code blocks.

What is the exact error message?

**Comment by marioricardojr** [comment]
sorry guy,
the error is this, I thank you if you can help me.
sorry for the inconvenience


gcc -std=c99 open62541.c ua_namespace_mtconnect.c  server.c -o server
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_60_begin’:
ua_namespace_mtconnect.c:2555:15: error: unknown type name ‘UA_CoordinateSystemTypeEnum’
 UA_STACKARRAY(UA_CoordinateSystemTypeEnum, variablenode_ns_1_i_6700_variant_Dat
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:2556:66: error: ‘UA_TYPES_COORDINATESYSTEMTYPEENUM’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_6700_variant_DataContents, &UA_TYPES[UA_TYPES_COORD
                                                                  ^
ua_namespace_mtconnect.c:2556:66: note: each undeclared identifier is reported only once for each function it appears in
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_117_begin’:
ua_namespace_mtconnect.c:7564:15: error: unknown type name ‘UA_AssetDescriptionType’
 UA_STACKARRAY(UA_AssetDescriptionType, variablenode_ns_1_i_9945_variant_DataCon
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:7565:66: error: ‘UA_TYPES_ASSETDESCRIPTIONTYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9945_variant_DataContents, &UA_TYPES[UA_TYPES_ASSET
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_118_begin’:
ua_namespace_mtconnect.c:7600:15: error: unknown type name ‘UA_GradeDataType’
 UA_STACKARRAY(UA_GradeDataType, variablenode_ns_1_i_9943_variant_DataContents, 
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:7601:66: error: ‘UA_TYPES_GRADEDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9943_variant_DataContents, &UA_TYPES[UA_TYPES_GRADE
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_119_begin’:
ua_namespace_mtconnect.c:7636:15: error: unknown type name ‘UA_IndexRangeDataType’
 UA_STACKARRAY(UA_IndexRangeDataType, variablenode_ns_1_i_9941_variant_DataConte
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:7637:66: error: ‘UA_TYPES_INDEXRANGEDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9941_variant_DataContents, &UA_TYPES[UA_TYPES_INDEX
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_142_begin’:
ua_namespace_mtconnect.c:9016:15: error: unknown type name ‘UA_OverlapDataType’
 UA_STACKARRAY(UA_OverlapDataType, variablenode_ns_1_i_9931_variant_DataContents
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:9017:66: error: ‘UA_TYPES_OVERLAPDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9931_variant_DataContents, &UA_TYPES[UA_TYPES_OVERL
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_143_begin’:
ua_namespace_mtconnect.c:9052:15: error: unknown type name ‘UA_OverlapDataType’
 UA_STACKARRAY(UA_OverlapDataType, variablenode_ns_1_i_9930_variant_DataContents
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:9053:66: error: ‘UA_TYPES_OVERLAPDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9930_variant_DataContents, &UA_TYPES[UA_TYPES_OVERL
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_145_begin’:
ua_namespace_mtconnect.c:9117:15: error: unknown type name ‘UA_CalibrationDateDataType’
 UA_STACKARRAY(UA_CalibrationDateDataType, variablenode_ns_1_i_9813_variant_Data
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:9118:66: error: ‘UA_TYPES_CALIBRATIONDATEDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9813_variant_DataContents, &UA_TYPES[UA_TYPES_CALIB
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_146_begin’:
ua_namespace_mtconnect.c:9153:15: error: unknown type name ‘UA_CalibrationDateDataType’
 UA_STACKARRAY(UA_CalibrationDateDataType, variablenode_ns_1_i_9812_variant_Data
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:9154:66: error: ‘UA_TYPES_CALIBRATIONDATEDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9812_variant_DataContents, &UA_TYPES[UA_TYPES_CALIB
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_147_begin’:
ua_namespace_mtconnect.c:9189:15: error: unknown type name ‘UA_CalibrationInitialsDataType’
 UA_STACKARRAY(UA_CalibrationInitialsDataType, variablenode_ns_1_i_9814_variant_
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:9190:66: error: ‘UA_TYPES_CALIBRATIONINITIALSDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9814_variant_DataContents, &UA_TYPES[UA_TYPES_CALIB
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_155_begin’:
ua_namespace_mtconnect.c:9416:15: error: unknown type name ‘UA_MaximumDataType’
 UA_STACKARRAY(UA_MaximumDataType, variablenode_ns_1_i_9937_variant_DataContents
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:9417:66: error: ‘UA_TYPES_MAXIMUMDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9937_variant_DataContents, &UA_TYPES[UA_TYPES_MAXIM
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_156_begin’:
ua_namespace_mtconnect.c:9452:15: error: unknown type name ‘UA_NominalDataType’
 UA_STACKARRAY(UA_NominalDataType, variablenode_ns_1_i_9939_variant_DataContents
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:9453:66: error: ‘UA_TYPES_NOMINALDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9939_variant_DataContents, &UA_TYPES[UA_TYPES_NOMIN
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_158_begin’:
ua_namespace_mtconnect.c:9517:15: error: unknown type name ‘UA_MaximumDataType’
 UA_STACKARRAY(UA_MaximumDataType, variablenode_ns_1_i_9933_variant_DataContents
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:9518:66: error: ‘UA_TYPES_MAXIMUMDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9933_variant_DataContents, &UA_TYPES[UA_TYPES_MAXIM
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_159_begin’:
ua_namespace_mtconnect.c:9553:15: error: unknown type name ‘UA_NominalDataType’
 UA_STACKARRAY(UA_NominalDataType, variablenode_ns_1_i_9935_variant_DataContents
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:9554:66: error: ‘UA_TYPES_NOMINALDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9935_variant_DataContents, &UA_TYPES[UA_TYPES_NOMIN
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_252_begin’:
ua_namespace_mtconnect.c:12350:15: error: unknown type name ‘UA_ToolLifeValueDataType’
 UA_STACKARRAY(UA_ToolLifeValueDataType, variablenode_ns_1_i_9926_variant_DataCo
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:12351:66: error: ‘UA_TYPES_TOOLLIFEVALUEDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9926_variant_DataContents, &UA_TYPES[UA_TYPES_TOOLL
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_253_begin’:
ua_namespace_mtconnect.c:12386:15: error: unknown type name ‘UA_ToolLifeValueDataType’
 UA_STACKARRAY(UA_ToolLifeValueDataType, variablenode_ns_1_i_9927_variant_DataCo
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:12387:66: error: ‘UA_TYPES_TOOLLIFEVALUEDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9927_variant_DataContents, &UA_TYPES[UA_TYPES_TOOLL
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_258_begin’:
ua_namespace_mtconnect.c:12635:15: error: unknown type name ‘UA_ToolLifeTypeEnum’
 UA_STACKARRAY(UA_ToolLifeTypeEnum, variablenode_ns_1_i_9924_variant_DataContent
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:12636:66: error: ‘UA_TYPES_TOOLLIFETYPEENUM’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9924_variant_DataContents, &UA_TYPES[UA_TYPES_TOOLL
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_277_begin’:
ua_namespace_mtconnect.c:13198:15: error: unknown type name ‘UA_CoordinateSystemTypeEnum’
 UA_STACKARRAY(UA_CoordinateSystemTypeEnum, variablenode_ns_1_i_7067_variant_Dat
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:13199:66: error: ‘UA_TYPES_COORDINATESYSTEMTYPEENUM’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_7067_variant_DataContents, &UA_TYPES[UA_TYPES_COORD
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_294_begin’:
ua_namespace_mtconnect.c:13863:15: error: unknown type name ‘UA_ToolLifeDirectionTypeEnum’
 UA_STACKARRAY(UA_ToolLifeDirectionTypeEnum, variablenode_ns_1_i_9925_variant_Da
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:13864:66: error: ‘UA_TYPES_TOOLLIFEDIRECTIONTYPEENUM’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9925_variant_DataContents, &UA_TYPES[UA_TYPES_TOOLL
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_297_begin’:
ua_namespace_mtconnect.c:13953:15: error: unknown type name ‘UA_MinimumDataType’
 UA_STACKARRAY(UA_MinimumDataType, variablenode_ns_1_i_9938_variant_DataContents
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:13954:66: error: ‘UA_TYPES_MINIMUMDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9938_variant_DataContents, &UA_TYPES[UA_TYPES_MINIM
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_298_begin’:
ua_namespace_mtconnect.c:13989:15: error: unknown type name ‘UA_MinimumDataType’
 UA_STACKARRAY(UA_MinimumDataType, variablenode_ns_1_i_9934_variant_DataContents
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:13990:66: error: ‘UA_TYPES_MINIMUMDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9934_variant_DataContents, &UA_TYPES[UA_TYPES_MINIM
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_302_begin’:
ua_namespace_mtconnect.c:14106:15: error: unknown type name ‘UA_FirmwareVersionDataType’
 UA_STACKARRAY(UA_FirmwareVersionDataType, variablenode_ns_1_i_9841_variant_Data
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:14107:66: error: ‘UA_TYPES_FIRMWAREVERSIONDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9841_variant_DataContents, &UA_TYPES[UA_TYPES_FIRMW
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_306_begin’:
ua_namespace_mtconnect.c:14328:15: error: unknown type name ‘UA_ItemIdDataType’
 UA_STACKARRAY(UA_ItemIdDataType, variablenode_ns_1_i_9942_variant_DataContents,
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:14329:66: error: ‘UA_TYPES_ITEMIDDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9942_variant_DataContents, &UA_TYPES[UA_TYPES_ITEMI
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_310_begin’:
ua_namespace_mtconnect.c:14458:15: error: unknown type name ‘UA_SignificantDigitsValueDataType’
 UA_STACKARRAY(UA_SignificantDigitsValueDataType, variablenode_ns_1_i_9880_varia
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:14459:66: error: ‘UA_TYPES_SIGNIFICANTDIGITSVALUEDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9880_variant_DataContents, &UA_TYPES[UA_TYPES_SIGNI
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_311_begin’:
ua_namespace_mtconnect.c:14494:15: error: unknown type name ‘UA_MeasurementValueDataType’
 UA_STACKARRAY(UA_MeasurementValueDataType, variablenode_ns_1_i_9887_variant_Dat
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:14495:66: error: ‘UA_TYPES_MEASUREMENTVALUEDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9887_variant_DataContents, &UA_TYPES[UA_TYPES_MEASU
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_312_begin’:
ua_namespace_mtconnect.c:14530:15: error: unknown type name ‘UA_MeasurementValueDataType’
 UA_STACKARRAY(UA_MeasurementValueDataType, variablenode_ns_1_i_9886_variant_Dat
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:14531:66: error: ‘UA_TYPES_MEASUREMENTVALUEDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9886_variant_DataContents, &UA_TYPES[UA_TYPES_MEASU
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_313_begin’:
ua_namespace_mtconnect.c:14566:15: error: unknown type name ‘UA_MeasurementValueDataType’
 UA_STACKARRAY(UA_MeasurementValueDataType, variablenode_ns_1_i_9885_variant_Dat
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:14567:66: error: ‘UA_TYPES_MEASUREMENTVALUEDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9885_variant_DataContents, &UA_TYPES[UA_TYPES_MEASU
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_314_begin’:
ua_namespace_mtconnect.c:14602:15: error: unknown type name ‘UA_CodeDataType’
 UA_STACKARRAY(UA_CodeDataType, variablenode_ns_1_i_9884_variant_DataContents, 1
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:14603:66: error: ‘UA_TYPES_CODEDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9884_variant_DataContents, &UA_TYPES[UA_TYPES_CODED
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_323_begin’:
ua_namespace_mtconnect.c:15039:15: error: unknown type name ‘UA_ComponentIdDataType’
 UA_STACKARRAY(UA_ComponentIdDataType, variablenode_ns_1_i_9868_variant_DataCont
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:15040:66: error: ‘UA_TYPES_COMPONENTIDDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9868_variant_DataContents, &UA_TYPES[UA_TYPES_COMPO
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_326_begin’:
ua_namespace_mtconnect.c:15129:15: error: unknown type name ‘UA_FeedrateValueDataType’
 UA_STACKARRAY(UA_FeedrateValueDataType, variablenode_ns_1_i_9917_variant_DataCo
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:15130:66: error: ‘UA_TYPES_FEEDRATEVALUEDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9917_variant_DataContents, &UA_TYPES[UA_TYPES_FEEDR
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_328_begin’:
ua_namespace_mtconnect.c:15198:15: error: unknown type name ‘UA_MinimumDataType’
 UA_STACKARRAY(UA_MinimumDataType, variablenode_ns_1_i_9915_variant_DataContents
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:15199:66: error: ‘UA_TYPES_MINIMUMDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9915_variant_DataContents, &UA_TYPES[UA_TYPES_MINIM
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_329_begin’:
ua_namespace_mtconnect.c:15234:15: error: unknown type name ‘UA_MaximumDataType’
 UA_STACKARRAY(UA_MaximumDataType, variablenode_ns_1_i_9914_variant_DataContents
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:15235:66: error: ‘UA_TYPES_MAXIMUMDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9914_variant_DataContents, &UA_TYPES[UA_TYPES_MAXIM
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_330_begin’:
ua_namespace_mtconnect.c:15270:15: error: unknown type name ‘UA_NominalDataType’
 UA_STACKARRAY(UA_NominalDataType, variablenode_ns_1_i_9916_variant_DataContents
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:15271:66: error: ‘UA_TYPES_NOMINALDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9916_variant_DataContents, &UA_TYPES[UA_TYPES_NOMIN
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_331_begin’:
ua_namespace_mtconnect.c:15306:15: error: unknown type name ‘UA_ConnectionCodeMachineSideDataType’
 UA_STACKARRAY(UA_ConnectionCodeMachineSideDataType, variablenode_ns_1_i_9918_va
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:15307:66: error: ‘UA_TYPES_CONNECTIONCODEMACHINESIDEDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9918_variant_DataContents, &UA_TYPES[UA_TYPES_CONNE
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_332_begin’:
ua_namespace_mtconnect.c:15342:15: error: unknown type name ‘UA_ProgramToolNumberDataType’
 UA_STACKARRAY(UA_ProgramToolNumberDataType, variablenode_ns_1_i_9905_variant_Da
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:15343:66: error: ‘UA_TYPES_PROGRAMTOOLNUMBERDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9905_variant_DataContents, &UA_TYPES[UA_TYPES_PROGR
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_333_begin’:
ua_namespace_mtconnect.c:15378:15: error: unknown type name ‘UA_NominalDataType’
 UA_STACKARRAY(UA_NominalDataType, variablenode_ns_1_i_9908_variant_DataContents
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:15379:66: error: ‘UA_TYPES_NOMINALDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9908_variant_DataContents, &UA_TYPES[UA_TYPES_NOMIN
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_334_begin’:
ua_namespace_mtconnect.c:15414:15: error: unknown type name ‘UA_MinimumDataType’
 UA_STACKARRAY(UA_MinimumDataType, variablenode_ns_1_i_9907_variant_DataContents
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:15415:66: error: ‘UA_TYPES_MINIMUMDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9907_variant_DataContents, &UA_TYPES[UA_TYPES_MINIM
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_335_begin’:
ua_namespace_mtconnect.c:15450:15: error: unknown type name ‘UA_MaximumDataType’
 UA_STACKARRAY(UA_MaximumDataType, variablenode_ns_1_i_9906_variant_DataContents
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:15451:66: error: ‘UA_TYPES_MAXIMUMDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9906_variant_DataContents, &UA_TYPES[UA_TYPES_MAXIM
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_337_begin’:
ua_namespace_mtconnect.c:15519:15: error: unknown type name ‘UA_ToolLifeValueDataType’
 UA_STACKARRAY(UA_ToolLifeValueDataType, variablenode_ns_1_i_9904_variant_DataCo
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:15520:66: error: ‘UA_TYPES_TOOLLIFEVALUEDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9904_variant_DataContents, &UA_TYPES[UA_TYPES_TOOLL
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_338_begin’:
ua_namespace_mtconnect.c:15555:15: error: unknown type name ‘UA_ToolLifeDirectionTypeEnum’
 UA_STACKARRAY(UA_ToolLifeDirectionTypeEnum, variablenode_ns_1_i_9902_variant_Da
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:15556:66: error: ‘UA_TYPES_TOOLLIFEDIRECTIONTYPEENUM’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9902_variant_DataContents, &UA_TYPES[UA_TYPES_TOOLL
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_339_begin’:
ua_namespace_mtconnect.c:15591:15: error: unknown type name ‘UA_ToolLifeTypeEnum’
 UA_STACKARRAY(UA_ToolLifeTypeEnum, variablenode_ns_1_i_9901_variant_DataContent
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:15592:66: error: ‘UA_TYPES_TOOLLIFETYPEENUM’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9901_variant_DataContents, &UA_TYPES[UA_TYPES_TOOLL
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_340_begin’:
ua_namespace_mtconnect.c:15627:15: error: unknown type name ‘UA_SerialNumberDataType’
 UA_STACKARRAY(UA_SerialNumberDataType, variablenode_ns_1_i_9870_variant_DataCon
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:15628:66: error: ‘UA_TYPES_SERIALNUMBERDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9870_variant_DataContents, &UA_TYPES[UA_TYPES_SERIA
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_344_begin’:
ua_namespace_mtconnect.c:15759:15: error: unknown type name ‘UA_ToolGroupDataType’
 UA_STACKARRAY(UA_ToolGroupDataType, variablenode_ns_1_i_9890_variant_DataConten
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:15760:66: error: ‘UA_TYPES_TOOLGROUPDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9890_variant_DataContents, &UA_TYPES[UA_TYPES_TOOLG
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_345_begin’:
ua_namespace_mtconnect.c:15795:15: error: unknown type name ‘UA_ToolIdDataType’
 UA_STACKARRAY(UA_ToolIdDataType, variablenode_ns_1_i_9889_variant_DataContents,
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:15796:66: error: ‘UA_TYPES_TOOLIDDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9889_variant_DataContents, &UA_TYPES[UA_TYPES_TOOLI
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_347_begin’:
ua_namespace_mtconnect.c:15858:15: error: unknown type name ‘UA_ReconditionCountValueDataType’
 UA_STACKARRAY(UA_ReconditionCountValueDataType, variablenode_ns_1_i_9898_varian
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:15859:66: error: ‘UA_TYPES_RECONDITIONCOUNTVALUEDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9898_variant_DataContents, &UA_TYPES[UA_TYPES_RECON
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_350_begin’:
ua_namespace_mtconnect.c:16084:15: error: unknown type name ‘UA_LocationsTypeEnum’
 UA_STACKARRAY(UA_LocationsTypeEnum, variablenode_ns_1_i_9929_variant_DataConten
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:16085:66: error: ‘UA_TYPES_LOCATIONSTYPEENUM’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9929_variant_DataContents, &UA_TYPES[UA_TYPES_LOCAT
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_352_begin’:
ua_namespace_mtconnect.c:16147:15: error: unknown type name ‘UA_LocusDataType’
 UA_STACKARRAY(UA_LocusDataType, variablenode_ns_1_i_9946_variant_DataContents, 
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:16148:66: error: ‘UA_TYPES_LOCUSDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9946_variant_DataContents, &UA_TYPES[UA_TYPES_LOCUS
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_354_begin’:
ua_namespace_mtconnect.c:16210:15: error: unknown type name ‘UA_ManufacturersDataType’
 UA_STACKARRAY(UA_ManufacturersDataType, variablenode_ns_1_i_9871_variant_DataCo
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:16211:66: error: ‘UA_TYPES_MANUFACTURERSDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9871_variant_DataContents, &UA_TYPES[UA_TYPES_MANUF
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_355_begin’:
ua_namespace_mtconnect.c:16246:15: error: unknown type name ‘UA_ManufacturersDataType’
 UA_STACKARRAY(UA_ManufacturersDataType, variablenode_ns_1_i_9944_variant_DataCo
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:16247:66: error: ‘UA_TYPES_MANUFACTURERSDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9944_variant_DataContents, &UA_TYPES[UA_TYPES_MANUF
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_359_begin’:
ua_namespace_mtconnect.c:16363:15: error: unknown type name ‘UA_LocationValueDataType’
 UA_STACKARRAY(UA_LocationValueDataType, variablenode_ns_1_i_9909_variant_DataCo
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:16364:66: error: ‘UA_TYPES_LOCATIONVALUEDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9909_variant_DataContents, &UA_TYPES[UA_TYPES_LOCAT
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_360_begin’:
ua_namespace_mtconnect.c:16399:15: error: unknown type name ‘UA_OverlapDataType’
 UA_STACKARRAY(UA_OverlapDataType, variablenode_ns_1_i_9912_variant_DataContents
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:16400:66: error: ‘UA_TYPES_OVERLAPDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9912_variant_DataContents, &UA_TYPES[UA_TYPES_OVERL
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_361_begin’:
ua_namespace_mtconnect.c:16435:15: error: unknown type name ‘UA_LocationsTypeEnum’
 UA_STACKARRAY(UA_LocationsTypeEnum, variablenode_ns_1_i_9910_variant_DataConten
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:16436:66: error: ‘UA_TYPES_LOCATIONSTYPEENUM’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9910_variant_DataContents, &UA_TYPES[UA_TYPES_LOCAT
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_362_begin’:
ua_namespace_mtconnect.c:16471:15: error: unknown type name ‘UA_OverlapDataType’
 UA_STACKARRAY(UA_OverlapDataType, variablenode_ns_1_i_9911_variant_DataContents
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:16472:66: error: ‘UA_TYPES_OVERLAPDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9911_variant_DataContents, &UA_TYPES[UA_TYPES_OVERL
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_364_begin’:
ua_namespace_mtconnect.c:16534:15: error: unknown type name ‘UA_MaximumCountDataType’
 UA_STACKARRAY(UA_MaximumCountDataType, variablenode_ns_1_i_9922_variant_DataCon
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:16535:66: error: ‘UA_TYPES_MAXIMUMCOUNTDATATYPE’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9922_variant_DataContents, &UA_TYPES[UA_TYPES_MAXIM
                                                                  ^
In file included from ua_namespace_mtconnect.h:12:0,
                 from ua_namespace_mtconnect.c:4:
ua_namespace_mtconnect.c: In function ‘function_ua_namespace_mtconnect_367_begin’:
ua_namespace_mtconnect.c:16635:15: error: unknown type name ‘UA_CutterStatusValueTypeEnum’
 UA_STACKARRAY(UA_CutterStatusValueTypeEnum, variablenode_ns_1_i_9896_variant_Da
               ^
open62541.h:159:42: note: in definition of macro ‘UA_STACKARRAY’
 # define UA_STACKARRAY(TYPE, NAME, SIZE) TYPE NAME[SIZE]
                                          ^
ua_namespace_mtconnect.c:16636:66: error: ‘UA_TYPES_CUTTERSTATUSVALUETYPEENUM’ undeclared (first use in this function)
 UA_init(variablenode_ns_1_i_9896_variant_DataContents, &UA_TYPES[UA_TYPES_CUTTE

**Comment by Pro** [comment]
You also need to include the MTConnect specific datatypes, i.e. generate them with:

```
python tools/generate_datatypes.py
-namespace=2
--type-csv=/open62541/deps/ua-nodeset/MTConnect/MTConnectModel.csv
--type-bsd=/open62541/deps/ua-nodeset/MTConnect/ 	Opc.Ua.MTConnect.Types.bsd
--no-builtin /open62541/src_mtconnect/ua_types_mtconnect
```
and then include the header and source file (`ua_types_mtconnect.h`, `ua_types_mtconnect.c`) in your code.

You then also need to change the following call to use the MTConnect array:
```
python tools/nodeset_compiler/nodeset_compiler.py
--internal-headers
--types-array=UA_TYPES
--types-array=UA_TYPES_MTCONNECT
--existing /open62541/deps/ua-nodeset/Schema/Opc.Ua.NodeSet2.xml
--xml /open62541/deps/ua-nodeset/MTConnect/Opc.Ua.MTConnect.NodeSet2.xml
/open62541/src_mtconnect/ua_namespace_mtconnect
```

**Comment by marioricardojr** [comment]
thanks @Pro , sorry for the inconvenience. you are the best

**Comment by Stasik0** [comment]
Can we get this into tutorials?

**Comment by Pro** [comment]
@marioricardojr no problem, we are here to help :) For the next issue please try to provide the detailed info in your first post so we can help you even faster.

@Stasik0 yes, would be nice. There are some more examples and docs missing for the NC. If I find some time, maybe I can extend it a bit.

**Comment by Stasik0** [comment]
I do not want to create additional load to you @Pro, maybe just an issue for grabs so we do not forget. Ideally there should be a copy-paste recipy how to get each companion compiled. Or maybe even cmake options as batteries-included.... wait, I create load again....
